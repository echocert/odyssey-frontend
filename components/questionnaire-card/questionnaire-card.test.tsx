import React from 'react';
import { create } from "react-test-renderer";
import { render, fireEvent, screen } from '@testing-library/react';
import QuestionnaireCard from './questionnaire-card';

describe('QuestionnaireCard component', () => {

    const testProps = {
        id: '1',
        title: 'Sample Questionnaire',
        name: 'Mike',
        formName:"Sample.pdf",
        onEdit: jest.fn(),
        onDelete: jest.fn(),
        onClone: jest.fn(),
        onShare: jest.fn(),
        onCardClick: jest.fn(),
    };

    it("Verify the snapshot of questionnaire card is matched", () => {
        const wrap = create(<QuestionnaireCard />).toJSON();
        expect(wrap).toMatchSnapshot();
    });

    it('Checks if default props are populated', () => {
        render(<QuestionnaireCard {...testProps} />);

            const titleText = screen.getByTestId('questionnaire_title');
            expect(testProps.formName).toEqual(titleText.textContent);
            
            // const nameText = screen.getByTestId('questionnaire_name');
            // expect(testProps.name).toEqual(nameText.textContent);
        });

 

    it('calls onEdit when the Edit icon is clicked', () => {
        render(<QuestionnaireCard {...testProps} />);
        const onEdit = screen.getByTestId('onEdit');
        fireEvent.click(onEdit);
        expect(testProps.onEdit).toHaveBeenCalledTimes(1);
    });
    it('calls onDelete when the Delete icon is clicked', () => {
        render(<QuestionnaireCard {...testProps} />);
        const onDelete = screen.getByTestId('onDelete');
        fireEvent.click(onDelete);
        expect(testProps.onDelete).toHaveBeenCalledTimes(1);
    });
    it('calls onClone when the clone icon is clicked', () => {
        render(<QuestionnaireCard {...testProps} />);
        const onClone = screen.getByTestId('onClone');
        fireEvent.click(onClone);
        expect(testProps.onClone).toHaveBeenCalled();
        const clone = screen.queryByTestId('delete');
        expect(clone).toBeDefined()
    });

    it('calls onCardClick when the card is clicked', () => {
        const { container } = render(<QuestionnaireCard {...testProps} />);
        const card = container.querySelector('.questionnaire-card');
        fireEvent.click(card);
        expect(testProps.onCardClick).toHaveBeenCalled();
    });
});
