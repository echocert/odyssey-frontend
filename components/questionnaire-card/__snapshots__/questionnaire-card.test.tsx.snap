// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionnaireCard component Verify the snapshot of questionnaire card is matched 1`] = `
<div>
  <div
    className="questionnaire-card"
    data-testid="card"
  >
    <div
      className="questionnaire-card__sets"
    >
      <div
        className="questionnaire-card__title"
        data-testid="questionnaire_title"
      >
        <label
          className="label-l4 label-header"
          data-testid="title"
          style={{}}
        />
      </div>
      <div
        className="questionnaire-card__heart-icon"
        onClick={[Function]}
      />
    </div>
    <div
      className="questionnaire-card__name"
      data-testid="questionnaire_name"
    >
      <label
        className="label-hero-l2 label-header-l2 label-l2 label-header"
        data-testid="name"
        style={{}}
      />
    </div>
    <div
      className="questionnaire-card__footer-icons"
      onClick={[Function]}
    >
      <div
        className="floating-menu-assign__questionnaire"
      >
        <div
          className="menu"
          data-testid="floating-menu"
        >
          <div
            className="menu__trigger"
            data-testid="floating-menu-trigger"
          >
            <div
              className="menu__anchor"
              data-testid="floating-menu-anchor"
            >
              <div
                className="questionnaire-card__icon questionnaire-card__footer-icons-email"
              >
                <test-file-stub
                  className="questionnaire-card__assign-icon"
                  height="20px"
                  width="20px"
                />
              </div>
              <div
                className="menu__content menu__content--left"
                data-testid="floating-menu-content"
                style={
                  {
                    "--tooltip-left": undefined,
                    "display": "",
                  }
                }
              >
                <div
                  className="menu__content-wrapper"
                >
                  <div
                    className="floating-menu-assign"
                  >
                    <label
                      className="label-hero-l2 label-header-l2 label-l2 label-header"
                      style={
                        {
                          "textTransform": "capitalize",
                        }
                      }
                    >
                      Assign Questionnaire
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="floating-menu-assign__questionnaire"
      >
        <div
          className="menu"
          data-testid="floating-menu"
        >
          <div
            className="menu__trigger"
            data-testid="floating-menu-trigger"
          >
            <div
              className="menu__anchor"
              data-testid="floating-menu-anchor"
            >
              <div
                className="questionnaire-card__icon questionnaire-card__footer-icons-send"
                data-testid="onEdit"
                onClick={[Function]}
              >
                <test-file-stub
                  height="20px"
                  width="20px"
                />
              </div>
              <div
                className="menu__content menu__content--left"
                data-testid="floating-menu-content"
                style={
                  {
                    "--tooltip-left": undefined,
                    "display": "",
                  }
                }
              >
                <div
                  className="menu__content-wrapper"
                >
                  <div
                    className="floating-menu-assign"
                  >
                    <label
                      className="label-hero-l2 label-header-l2 label-l2 label-header"
                      style={
                        {
                          "textTransform": "capitalize",
                        }
                      }
                    >
                      Edit Questionnaire
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="floating-menu-assign__questionnaire"
      >
        <div
          className="menu"
          data-testid="floating-menu"
        >
          <div
            className="menu__trigger"
            data-testid="floating-menu-trigger"
          >
            <div
              className="menu__anchor"
              data-testid="floating-menu-anchor"
            >
              <div
                className="questionnaire-card__icon questionnaire-card__footer-icons-delete"
              >
                <test-file-stub
                  data-testid="onDelete"
                  height="20px"
                  width="20px"
                />
              </div>
              <div
                className="menu__content menu__content--left"
                data-testid="floating-menu-content"
                style={
                  {
                    "--tooltip-left": undefined,
                    "display": "",
                  }
                }
              >
                <div
                  className="menu__content-wrapper"
                >
                  <div
                    className="floating-menu-assign"
                  >
                    <label
                      className="label-hero-l2 label-header-l2 label-l2 label-header"
                      style={
                        {
                          "textTransform": "capitalize",
                        }
                      }
                    >
                      Delete Questionnaire
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="floating-menu-assign__questionnaire"
      >
        <div
          className="menu"
          data-testid="floating-menu"
        >
          <div
            className="menu__trigger"
            data-testid="floating-menu-trigger"
          >
            <div
              className="menu__anchor"
              data-testid="floating-menu-anchor"
            >
              <div
                className="questionnaire-card__icon questionnaire-card__footer-icons-clone"
                data-testid="onClone"
              >
                <test-file-stub
                  height="20px"
                  width="20px"
                />
              </div>
              <div
                className="menu__content menu__content--left"
                data-testid="floating-menu-content"
                style={
                  {
                    "--tooltip-left": undefined,
                    "display": "",
                  }
                }
              >
                <div
                  className="menu__content-wrapper"
                >
                  <div
                    className="floating-menu-assign"
                  >
                    <label
                      className="label-hero-l2 label-header-l2 label-l2 label-header"
                      style={
                        {
                          "textTransform": "capitalize",
                        }
                      }
                    >
                      Clone Questionnaire
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
